L1Trigger/CSCTriggerPrimitives/src/CSCGEMMotherboardME11.cc

```diff
@@ -0,0 +1,855 @@
+            n1a++;
+            if (infoV > 0) LogDebug("CSCGEMMotherboardME11")
+			     << "1a LCT"<<i+1<<" "<<bx<<"/"<<cbx<<": "<<allLCTs1a(bx,mbx,i)<< std::endl;
+          }
+        }
+      if (infoV > 0 and n1a+n1b>0) LogDebug("CSCGEMMotherboardME11")
+				     <<"bx "<<bx<<" nnLCT:"<<n1a<<" "<<n1b<<" "<<n1a+n1b<<std::endl;
+    } // x-bx sorting
+
+    // Maximum 2 per whole ME11 per BX case:
+    // (supposedly, now we should have max 2 per bx in each 1a and 1b)
+    if (n1a+n1b > max_lcts and tmb_cross_bx_algo == 1)
+    {
+      // do it simple so far: take all low eta 1/b stubs
+      unsigned int nLCT=n1b;
+      n1a=0;
+      // right now nLCT<=2; cut 1a if necessary
+      for (unsigned int mbx=0; mbx<match_trig_window_size; mbx++)
+        for (int i=0;i<CSCConstants::MAX_LCTS_PER_CSC;i++)
+          if (allLCTs1a(bx,mbx,i).isValid()) {
+            nLCT++;
+            if (nLCT>max_lcts) allLCTs1a(bx,mbx,i).clear();
+            else n1a++;
+          }
+      if (infoV > 0 and nLCT>0) LogDebug("CSCGEMMotherboardME11")
+				  <<"bx "<<bx<<" nnnLCT: "<<n1a<<" "<<n1b<<" "<<n1a+n1b<< std::endl;
+    }
+  }// reduction per bx
+
+  bool first = true;
+  unsigned int n1b=0, n1a=0;
+  LogTrace("CSCGEMCMotherboardME11") << "========================================================================" << std::endl
+                                     << "Counting the final LCTs" << std::endl
+                                     << "========================================================================" << std::endl
+                                     << "tmb_cross_bx_algo: " << tmb_cross_bx_algo << std::endl;
+
+  for (const auto& p : readoutLCTs1b()) {
+    n1b++;
+    if (debug_matching)
+      LogTrace("CSCGEMCMotherboardME11") << "1b LCT "<<n1b<<"  " << p <<std::endl;
+  }
+
+  for (const auto& p : readoutLCTs1a()){
+    n1a++;
+    if (debug_matching)
+      LogTrace("CSCGEMCMotherboardME11") << "1a LCT "<<n1a<<"  " << p <<std::endl;
+  }
+}
+
+std::vector<CSCCorrelatedLCTDigi> CSCGEMMotherboardME11::readoutLCTs1a() const
+{
+  return readoutLCTs(ME1A);
+}
+
+
+std::vector<CSCCorrelatedLCTDigi> CSCGEMMotherboardME11::readoutLCTs1b() const
+{
+  return readoutLCTs(ME1B);
+}
+
+
+// Returns vector of read-out correlated LCTs, if any.  Starts with
+// the vector of all found LCTs and selects the ones in the read-out
+// time window.
+std::vector<CSCCorrelatedLCTDigi> CSCGEMMotherboardME11::readoutLCTs(enum CSCPart me1ab) const
+{
+  std::vector<CSCCorrelatedLCTDigi> tmpV;
+
+  // The start time of the L1A*LCT coincidence window should be related
+  // to the fifo_pretrig parameter, but I am not completely sure how.
+  // Just choose it such that the window is centered at bx=7.  This may
+  // need further tweaking if the value of tmb_l1a_window_size changes.
+  //static int early_tbins = 4;
+  // The number of LCT bins in the read-out is given by the
+  // tmb_l1a_window_size parameter, forced to be odd
+  static int lct_bins   =
```

Comment:
This is not thread safe

Reply:
apologies - I missed this - it looks like the static can just be an int instead (in both cases) @dildick